server-common/src/main/java/org/apache/kafka/server/group/share/NoOpShareStatePersister.java

```diff
@@ -26,10 +26,23 @@
 import java.util.concurrent.CompletableFuture;
 import java.util.stream.Collectors;
 
+/**
+ * A no-op singleton implementation of {@link Persister} interface.
+ */
 public class NoOpShareStatePersister implements Persister {
 
   private static final Logger log = LoggerFactory.getLogger(NoOpShareStatePersister.class);
+  private static Persister instance = null;
+
+  private NoOpShareStatePersister() {
+  }
 
+  public static synchronized Persister getInstance() {
+    if (instance == null) {
+      instance = new NoOpShareStatePersister();
+    }
+    return instance;
+  }
```

Comment:
Can we have double check lock instead of method synchronization because I can see this code will be executed always when share partition is created. Though we can also optimize that i.e. just call the `NoOpShareStatePersister.getInstance()` only when the config is not provided.

Reply:
I see your point. Also, one more point that I missed was that I am creating multiple instances of SharePersisterClass for different share partitions (when they are initialized) in case the config is provided. I think the better logic is to have a global Persister object in SPM and initialize it using the config within the constructor itself

Reply:
Yes you are right.